frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/EventListenerTriggers.tsx

```diff
@@ -24,7 +24,7 @@ const EventListenerTriggers: React.FC<EventListenerTriggersProps> = ({ namespace
       <dd>
         {triggers.map((trigger) => {
           const triggerTemplateKind = referenceForModel(TriggerTemplateModel);
-          const triggerTemplateName = trigger.template.name;
+          const triggerTemplateName = trigger.template?.name;
```

Comment:
If you don't have a template name, I think we need to return `null` to just kill this node. No sense in trying to craft a broken block.

Reply:
filtering triggers based on template?.name and returning null early on in the component, so that <dt>Tiggers <dt> is also not rendered.

Reply:
I think there was a misunderstanding here.

You made this `trigger.template?.name` optional, but it will fail to render properly if the `triggerTemplateName` is `undefined`. So we have to do something about the iterations of this map that do NOT have a `triggerTemplateName`. 

My recommendation is you return `null` and drop this iteration from rendering a broken link. I'm open to other suggestions, but we can't just leave it.

Reply:
> You made this trigger.template?.name optional, but it will fail to render properly if the triggerTemplateName is undefined. So we have to do something about the iterations of this map that do NOT have a triggerTemplateName.

I am filtering out the tirggers based on tempate.name on the[ top of this component](https://github.com/openshift/console/pull/7889/files#diff-b7d6724fd77af9ccf450c9cee092536c0175d346fb63fe6ce93fce8874112eebR21-R24) now. so the map will only contain the set of triggers that consists of template.name.